### PR TITLE
IDE-593 Folder icons don't open and close correctly

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-593

Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review. I changed the folder behavoir slightly and to be more intuitive. Now, the folder icons only open when a tree branch is opened using the "+" or double-click. Selection only keeps the folder icons closed. Also, I changed the "...Loading..." to ">> empty <<" for empty folders and keep them open. This solved a logical problem with open and closed folders for the workunits. To tedious to explain the details but it turns out that the "empty" solution is much more intuitive then what is currently implemented which doesn't open the folder if there is nothing there. This feels more final and definite. Let me know if what you think about the new behavior.

File: wlib/RepositoryTreeNode.cpp [hunk 1]
```diff
@@ -177,18 +177,34 @@ void CModuleNode::ItemExpanding()
 	else
 		m_module->GetAttributes(attributes);
 	m_owner->OnLoaded(m_module, attributes);
+
+	CLoadingNode * loadingNode = new CLoadingNode();
+	loadingNode = new CLoadingNode();
```
Comment (GordonSmith) [inline wlib/RepositoryTreeNode.cpp]: Duplicate ```new CLoadingNode```

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you

**Comment by dehilsterlexis** [comment]
@GordonSmith poking you to review.

**Pull request merged**